tests/integration/targets/lookup_passwordstore/tasks/tests.yml

```diff
@@ -151,3 +168,86 @@
   assert:
     that:
       - readyamlpass == 'testpassword\nrandom additional line'
+
+- name: Create a password in a folder
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'folder/test-pass length=8 create=yes') }}"
+
+- name: Fetch password from folder
+  set_fact:
+    readpass: "{{ lookup('community.general.passwordstore', 'folder/test-pass') }}"
+
+- name: Verify password from folder
+  assert:
+    that:
+      - readpass == newpass
+
+- name: Try to read folder as passname
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'folder') }}"
+  ignore_errors: true
+  register: eval_error
+
+- name: Make sure reading folder as passname failed
+  assert:
+    that:
+      - eval_error is failed
+      - "\"passname folder not found\" in eval_error.msg"
+
+- name: Change passwordstore location explicitly
+  set_fact:
+    passwordstore: "{{ lookup('env','HOME') }}/.password-store"
+
+- name: Make sure password store still works with explicit location set
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'test-pass') }}"
+
+- name: Change passwordstore location to a non-existent place
+  set_fact:
+    passwordstore: "somenonexistentplace"
+
+- name: Try reading from non-existent passwordstore location
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'test-pass') }}"
+  ignore_errors: true
+  register: eval_error
+
+- name: Make sure reading from non-existent passwordstore location failed
+  assert:
+    that:
+      - eval_error is failed
+      - "\"Passwordstore directory 'somenonexistentplace' does not exist\" in eval_error.msg"
+
+- name: Move original pass out of the way
+  command:
+    argv:
+      - mv
+      - "{{ passpath }}"
+      - "{{ passpath }}.testorig"
+  args:
+    creates: "{{ passpath }}.testorig"
+
+- name: Create dummy pass script
```

Comment:
How about putting this and all following tasks into a `block`, with the last one (`Move original pass back into place`) being in the `always:` part? That way this block will try to clean itself up, but if it failed the `Move original pass into place if there was a leftover` task at the top will clean it up before running the tests.

Reply:
Good idea